core/src/main/scala/org/apache/spark/status/KVUtils.scala

```diff
@@ -80,6 +89,44 @@ private[spark] object KVUtils extends Logging {
     db
   }
 
+  def createKVStore(
+      storePath: Option[File],
+      diskBackend: HybridStoreDiskBackend.Value,
+      conf: SparkConf): KVStore = {
+    storePath.map { path =>
+      val dir = diskBackend match {
+        case LEVELDB => "listing.ldb"
+        case ROCKSDB => "listing.rdb"
+      }
+
+      val dbPath = Files.createDirectories(new File(path, dir).toPath()).toFile()
+      Utils.chmod700(dbPath)
+
+
+      val metadata = FsHistoryProviderMetadata(
+        FsHistoryProvider.CURRENT_LISTING_VERSION,
+        AppStatusStore.CURRENT_VERSION,
+        conf.get(History.HISTORY_LOG_DIR))
+
+      try {
+        open(dbPath, metadata, conf, Some(diskBackend))
+      } catch {
```

Comment:
For Live UI, are these two exception scenarios possible?

Reply:
No. The LevelDB exception is not possible either. There are only two error-handling branches, and seems too much to distinguish whether the caller is for live UI or SHS.

Reply:
If unexpected data cleaning doesn't occur in the live UI scenario, I think it is OK

Reply:
If something goes wrong and the kv store can't be created for live ui then the entire application fails?

Reply:
@tgravescs If there is exception from RocksDB, it will delete the target directory and open the KV store again. Otherwise if the error is not from RocksDB, it will fall back to the in-memory store.
It follows the same logic in SHS here.